blocks/utils.go

```diff
@@ -97,18 +97,27 @@ func getProposalMetadata(
 	}
 	majority := cfg.OptFilterThreshold * len(proposals)
 	var majorityState *meshState
+	var highest int
 	for _, ms := range meshHashes {
 		logger.With().Debug("mesh hash",
 			ms.hash,
 			log.Int("count", ms.count),
 			log.Int("threshold", cfg.OptFilterThreshold),
 			log.Int("num_proposals", len(proposals)))
-		if ms.hash != types.EmptyLayerHash && ms.count*100 >= majority {
-			majorityState = ms
+		if ms.hash != types.EmptyLayerHash {
+			if ms.count > highest {
+				highest = ms.count
+			}
+			if ms.count*100 >= majority {
+				majorityState = ms
+				optimistic.Inc()
+				agreement.Set(float64(ms.count*100) / float64(len(proposals)))
+			}
```

Comment:
i don't think that updating metrics in util function is an ok approach.

also storing last value is somewhat useless

Reply:
maybe i misunderstand the metrics types.
my main goal was to see the degree of consensus in each layer.
layer 19 -> 95%
layer 20 -> 100%
layer 21 -> 92%
and see the up and down trend

Reply:
i don't know if this will work like that, it depends on scraping timings and prometheus queuing. but i don't mind if you want to check.

i would agreement fraction in histogram with 10 buckets (0-10, 10-20, and so on)